linkerd/app/core/src/errors.rs

```diff
@@ -229,32 +235,81 @@ impl<RspB: Default + hyper::body::HttpBody> respond::Respond<http::Response<RspB
                     return Ok(rsp);
                 }
 
-                let status = http_status(&*error);
-                debug!(%status, version = ?self.version, "Handling error with HTTP response");
-                Ok(http::Response::builder()
+                let rsp = set_http_status(builder, &*error)
                     .version(self.version)
-                    .status(status)
                     .header(http::header::CONTENT_LENGTH, "0")
                     .body(ResponseBody::default())
-                    .expect("error response must be valid"))
+                    .expect("error response must be valid");
+                let status = rsp.status();
+                debug!(%status, version = ?self.version, "Handling error with HTTP response");
+                Ok(rsp)
             }
         }
     }
 }
 
-fn http_status(error: &(dyn std::error::Error + 'static)) -> StatusCode {
+fn set_l5d_error_header(error: &(dyn std::error::Error + 'static)) -> http::response::Builder {
+    let mut builder = http::Response::builder();
```

Comment:
nit: it would be a bit more consistent if this took the builder as an argument, like `set_http_status` does. and its name (`set_...`) indicates that it sets a header, not that it creates a new response builder.